enterprise/coderd/coderd.go

```diff
@@ -287,6 +287,15 @@ func New(ctx context.Context, options *Options) (_ *API, err error) {
 				r.Get("/", api.groupByOrganization)
 			})
 		})
+		r.Route("/organizations/{organization}/users/{user}/reduced-groups", func(r chi.Router) {
+			r.Use(
+				apiKeyMiddleware,
+				api.templateRBACEnabledMW,
+				httpmw.ExtractOrganizationParam(api.Database),
+				httpmw.ExtractUserParam(options.Database),
+			)
+			r.Get("/", api.reducedGroupsByUserAndOrganization)
+		})
```

Comment:
I feel like this is a good candidate to make a more generalized `/groups` database query, and move this particular route to `GET /users/{user}/groups` or something like that.

The current settings page is using `/organizations/{organization}/groups` for a page that should show all groups. So the frontend is incorrect when multi-org is released ([it is in progress, and I just made an issue for this after seeing this PR!](https://github.com/coder/coder/issues/14213)).

-----

Basically there exists a reason to query groups by `user_id`, and a reason to query them by `organization_id`, and maybe both. We should have a database function that takes various filter arguments, so we can use the same backend call.

Not necessary to do in this PR though, it can be a follow up.

-------

For this PR, at the very least, we should move this to the existing `/groups` endpoint on organizations, and just add a query param to filter by user imo. So like:

`/organizations/{organization}/groups/{groupName}?with_member=<user_id>` or something like that? 


https://github.com/coder/coder/blob/main/enterprise/coderd/groups.go#L348-L352

The returned `codersdk.Group` will just have a potentially empty (or 1 element, yourself) `Members` field. And just add `TotalMemberCount` to the existing `codersdk.Group`.

Reply:
Once we add the `TotalMemberCount` to `codersdk.Group` this route isn't needed anymore and we can keep on using the existing `/organizations/{organization}/groups` on the account page. To keep this PR focused, I think any new endpoints would better be added in a PR related to https://github.com/coder/coder/issues/14213.

Reply:
@hugodutka happy to just solve the permissions here, and do endpoints after :+1: